File: frontend/gameapp-frontend/src/domain/value-objects/scenes/change-biome-scene.js
import { Scene } from "./scene";

// ChangeBiomeScene
export class ChangeBiomeScene extends Scene {
    constructor(name, description, biome) {
        super(name, description, biome);
    }

    // setter
    setSceneName(newName) { return new ChangeBiomeScene(newName, this._description, this._biome); }
    setSceneDescription(newDescription) { return new ChangeBiomeScene(this._name, newDescription, this._biome); }
    setBiome(newBiome) { return new ChangeBiomeScene(this._name, this._description, newBiome); }



    toString() {
        return `ChangeBiomeScene: ${this._name} - Description=${this._description}, BiomeToChange=${this._biome}`;
    }
}

File: backend/GameApp.Domain/Entities/Scenes/ChangeBiomeScene.cs
using GameApp.Domain.Enumerates;
using GameApp.Domain.ValueObjects.Scenes;

namespace GameApp.Domain.Entities.Scenes;

// Class for scene where the biome changes in the biome atribute
public class ChangeBiomeScene : Scene
{

    // Constructor
    public ChangeBiomeScene(SceneName name, SceneDescription description, Biome biome)
        : base(name, description, biome)
    {
    }

    // Restore Constructor
    public ChangeBiomeScene(Guid id, SceneName name, SceneDescription description, Biome biome)
        : base(id, name, description, biome)
    {
    }

    // Setters 
    public ChangeBiomeScene SetSceneName(SceneName newName) =>
    new ChangeBiomeScene(GetGuid(), newName, GetDescription(), GetBiome());

    public ChangeBiomeScene SetSceneDescription(SceneDescription newDescription) =>
    new ChangeBiomeScene(GetGuid(), GetName(), newDescription, GetBiome());

    public ChangeBiomeScene SetBiome(Biome newBiome) =>
    new ChangeBiomeScene(GetGuid(), GetName(), GetDescription(), newBiome);

    // To string
    public override string ToString() =>
        $"ChangeBiomeScene({GetGuid()}): {GetName()} - {GetDescription()}, BiomeToChange={GetBiome()}";
}
